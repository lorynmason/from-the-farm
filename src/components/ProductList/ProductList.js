import React from 'react';
import Product from '../Product/Product';
import PropTypes from 'prop-types';

export const ProductList = ({ products }) => {
  let productsToReturn;
  if (products) {
    productsToReturn = products.map((product) => {
      return <Product product={product} key={product.id} />
    });
  }
  return (
    <section className="table-container">
      <table className="product-list">
        <thead className="product product-list-header">
          <tr>
            <th>Product</th>
            <th>Vendor</th>
            <th>Description</th>
            <th>$Price/Unit</th>
          </tr>
        </thead>
        <tbody>
          {productsToReturn}
        </tbody>
      </table>
    </section>
  )
}

ProductList.propTypes = {
  products: PropTypes.array
}
