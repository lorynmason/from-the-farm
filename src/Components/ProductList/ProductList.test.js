import React from 'react';
import ProductList from './ProductList';
import { shallow } from 'enzyme'

describe('ProductList', () => {
  it('should match snapshot', () => {
    let wrapper = shallow(<ProductList />)
    expect(wrapper).toMatchSnapshot()
  });
})