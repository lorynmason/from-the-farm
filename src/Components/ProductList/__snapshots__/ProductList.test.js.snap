// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Product
          product={
            Object {
              "description": "Pueblo hot roasted",
              "name": "chile pepper",
              "price": 20,
              "quantity": 1,
              "unit": "bushel",
            }
          }
        />,
        <Product
          product={
            Object {
              "description": "mini washed",
              "name": "carrot",
              "price": 5,
              "quantity": 10,
              "unit": "lbs",
            }
          }
        />,
        <Product
          product={
            Object {
              "description": "world famous Rocky Ford",
              "name": "cantaloupe",
              "price": 15,
              "quantity": 70,
              "unit": "lbs",
            }
          }
        />,
      ],
      "className": "product-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "product": Object {
            "description": "Pueblo hot roasted",
            "name": "chile pepper",
            "price": 20,
            "quantity": 1,
            "unit": "bushel",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "product": Object {
            "description": "mini washed",
            "name": "carrot",
            "price": 5,
            "quantity": 10,
            "unit": "lbs",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "product": Object {
            "description": "world famous Rocky Ford",
            "name": "cantaloupe",
            "price": 15,
            "quantity": 70,
            "unit": "lbs",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Product
            product={
              Object {
                "description": "Pueblo hot roasted",
                "name": "chile pepper",
                "price": 20,
                "quantity": 1,
                "unit": "bushel",
              }
            }
          />,
          <Product
            product={
              Object {
                "description": "mini washed",
                "name": "carrot",
                "price": 5,
                "quantity": 10,
                "unit": "lbs",
              }
            }
          />,
          <Product
            product={
              Object {
                "description": "world famous Rocky Ford",
                "name": "cantaloupe",
                "price": 15,
                "quantity": 70,
                "unit": "lbs",
              }
            }
          />,
        ],
        "className": "product-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "product": Object {
              "description": "Pueblo hot roasted",
              "name": "chile pepper",
              "price": 20,
              "quantity": 1,
              "unit": "bushel",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "product": Object {
              "description": "mini washed",
              "name": "carrot",
              "price": 5,
              "quantity": 10,
              "unit": "lbs",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "product": Object {
              "description": "world famous Rocky Ford",
              "name": "cantaloupe",
              "price": 15,
              "quantity": 70,
              "unit": "lbs",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
